### PR TITLE
Add integration test for jwt authentication

### PR DESCRIPTION
**Pull request by pitlv2109**
#14196

### CODE REVIEW

File: tests/integration/security/authn/jwt/jwt_test.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+			ns := namespace.NewOrFail(t, ctx, "authn-jwt", true)
+
+			var a, b, c echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				With(&c, util.EchoConfig("c", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			testCases := []struct {
```
Comment (yangminzhu) [inline tests/integration/security/authn/jwt/jwt_test.go]: Can we just first migrate the existing jwt e2e tests to the new framework? https://github.com/istio/istio/blob/83102c738251e2bfdb8cf18002d1ca4c674a0aa5/tests/e2e/tests/pilot/authn_policy_test.go#L101

This helps to make sure we have the same coverage in the 2 test framework, and we can add more tests (only in the new framework) after.

Comment (pitlv2109) [reply tests/integration/security/authn/jwt/jwt_test.go]: Tests from the old framework are also run besides in the new framework. Also, the jwt test here https://github.com/istio/istio/blob/49c978439ddde62f4d5e0cf672976294797cf2fa/tests/e2e/tests/pilot/testdata/authn/v1alpha1/authn-policy-jwt.yaml.tmpl is pretty basic. I think this new one cover more cases, but I can look into the old test as a follow up PR if you think we need to

Comment (yangminzhu) [reply tests/integration/security/authn/jwt/jwt_test.go]: The point is to first have same coverage then increase more cases, so when we later remove the old e2e test, we don't worry about losing any coverage, and the easier way to do this is starting from the same test in the new test framework.

One thing not covered in the new test framework, is the case for https://github.com/istio/istio/blob/83102c738251e2bfdb8cf18002d1ca4c674a0aa5/tests/e2e/tests/pilot/authn_policy_test.go#L153, note we specify two jwt issuers in the same jwt policy and make sure only it's used in different paths at the same time.

Comment (pitlv2109) [reply tests/integration/security/authn/jwt/jwt_test.go]: Is that the only difference (besides the ingress one)? If so, I can ask that test case to this PR.

File: tests/integration/security/authn/jwt/jwt_test.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	authHeaderKey = "Authorization"
+	validJwt      = "eyJhbGciOiJSUzI1NiIsImtpZCI6IkRIRmJwb0lVcXJZOHQyenBBMnFYZkNtcjVWTzVaRX" +
```
Comment (quanjielin) [inline tests/integration/security/authn/jwt/jwt_test.go]: move to a file instead putting in .go

File: tests/integration/security/authn/jwt/jwt_test.go (hunk 1 above)
Comment (yangminzhu) [reply tests/integration/security/authn/jwt/jwt_test.go]: I'm not sure, you need to look into the code, that's what I mean it's harder to figure out if we start by adding new tests, anyway, you can choose either of the following:
- Add new tests and then migrate the old tests
- Migrate old tests and then add new tests
Either way should make sure when we later remove the old test code, we will have the same coverage in the new test.

File: tests/integration/security/authn/jwt/jwt_test.go (hunk 2 above)
Comment (yangminzhu) [reply tests/integration/security/authn/jwt/jwt_test.go]: please also add a comment about the contents of the JWT token, like here https://github.com/istio/istio/blob/83102c738251e2bfdb8cf18002d1ca4c674a0aa5/tests/e2e/tests/pilot/authn_policy_test.go#L103

File: tests/integration/security/authn/jwt/jwt_test.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+							From: a,
+							Options: echo.CallOptions{
+								Target:   b,
+								PortName: "http",
+								Scheme:   scheme.HTTP,
+								Headers: map[string][]string{
+									authHeaderKey: {"Bearer " + expiredJwt},
+								},
+							},
+							ExpectSuccess: false,
```
Comment (quanjielin) [inline tests/integration/security/authn/jwt/jwt_test.go]: are we able to get the respcode(200/401) and verify ?

**Comment by howardjohn** [comment]
@pitlv2109 can you move this to master? See https://discuss.istio.io/t/1-2-release-branch-cut/2372/2

**Comment by pitlv2109** [comment]
@howardjohn  Sure thing. Thanks for the link :)

File: tests/integration/security/authn/jwt/jwt_test.go (hunk 2 above)
Comment (pitlv2109) [reply tests/integration/security/authn/jwt/jwt_test.go]: I think we can just do in line there and add some description like Yangmin suggested. I don't see the need of putting it in a file and do open/read content from file.

File: tests/integration/security/authn/jwt/jwt_test.go (hunk 3 above)
Comment (pitlv2109) [reply tests/integration/security/authn/jwt/jwt_test.go]: Good point. The checking function can check for success (200), but failure does not mean 401. I forgot to update that function :D

**Comment by pitlv2109** [comment]
Close this one and will open another one in `master`.
/close

**Pull request closed without merging**